tests/test_basics.py

```diff
@@ -240,23 +240,6 @@ def test_client_initialized_within_scope(sentry_init, caplog):
     assert record.msg.startswith("init() called inside of pushed scope.")
 
 
-def test_scope_leaks_cleaned_up(sentry_init, caplog):
-    caplog.set_level(logging.WARNING)
-
-    sentry_init(debug=True)
-
-    old_stack = list(Hub.current._stack)
-
-    with push_scope():
-        push_scope()
```

Comment:
I definitely understand that scope cleanup needs to happen in async envs. However,  if it is as critical as you say that push_scope() continues working the "eager" way it does now AND have it work as a context manager AND like this
```python
@sentry_sdk.push_scope()
def foo():
    sentry_sdk.set_tag("key", "value")
```
Then I haven't thought of how to make it work, it's not possible, or something else may have to give. From my outsider's perspective the easiest solution is to deprecate push_scope(callback) functionality + alter the Hub and ScopeManager classes so that the following examples would all work:
```python
@sentry_sdk.push_scope
def foo():
    sentry_sdk.set_tag("key", "value")


with sentry_sdk.push_scope():
    print("pushed scope!")


scope = sentry_sdk.push_scope()
```
but... that feels like a very major change to the api.